edb/pgsql/compiler/relctx.py

```diff
@@ -1157,11 +1199,54 @@ def rel_join(
             type=join_type, larg=larg, rarg=rarg, quals=condition)
 
 
+def _lateral_union_join(
+    query: pgast.SelectStmt,
+    right_rvar: pgast.RangeSubselect,
+    *,
+    ctx: context.CompilerContextLevel,
+) -> None:
+    def _inject_filter(component: pgast.Query) -> None:
+        condition = None
+
+        for path_id in right_rvar.query.path_scope:
+            lref = maybe_get_path_var(
```

Comment:
Do you think it's worth trying to unify `_inject_filter` and the main logic in `_plain_join`?

Reply:
They are subtly different, I don't think unifying them would help readability.